students/mattmaeda/stockscreener/exchange/views.py

```diff
@@ -0,0 +1,266 @@
+"""
+Contains the view logic for stockscreener app
+"""
+import datetime
+import iexfinance
+from django.shortcuts import render
+from django.http import HttpResponse
+from django.urls import reverse
+from requests import ConnectionError
+from iexfinance import Stock
+from .models import Screen
+
+# Create your views here.
+def index(request):
+    """ Main page """
+    message ="Pleast enter a stock symbol (e.g., GOOG)"
+    response = None
+
+    if request.method == "POST":
+
+        try:
+            ticker = request.POST.get("ticker", "")
+            symbol = ticker.upper()
+            company = get_stock_info(symbol)
+
+            if company is not None:
+                history = get_price_history(symbol)
+                response = run_screens(history)
+                message = "Here are your results for {}".format(symbol)
+            else:
+                message = "Invalid stock symbol '{}'".format(symbol)
```

Comment:
There is a lot of code in this try block. Think about where you expect to get the ConnectionError, and try to limit the block in the try block to just that. Especially try to avoid conditionals in a try block.